xun/functions/store/disk.py

```diff
@@ -0,0 +1,121 @@
+from .store import Store
+from .store import StoreDriver
+from pathlib import Path
+import hashlib
+import pickle
+
+
+def key_hash(key):
+    pickled = pickle.dumps(key)
+    return hashlib.sha256(pickled).hexdigest()
+
+
+class Disk(Store):
+    class Driver(StoreDriver):
+        def __init__(self, dir):
+            self.dir = dir
+            self.index = {}
+
+            (self.dir / 'keys').mkdir(parents=True, exist_ok=True)
+            (self.dir / 'values').mkdir(parents=True, exist_ok=True)
+
+        def refresh_index(self):
+            files = [
+                p for p in (self.dir / 'keys').iterdir()
+                if p.is_file() and p.suffix != '.tmp'
+            ]
+            for path in files:
+                if path.name not in self.index:
+                    with open(str(path.resolve()), 'rb') as f:
+                        try:
+                            key = pickle.load(f)
+                        except EOFError:
+                            # This can occur if the index file is currently
+                            # being written to somewhere else on the network.
+                            # We consider these as not being part of the index.
+                            continue
+
+                    self.index[path.name] = key
+
+                    if __debug__ or True:
```

Comment:
Won't this always be True? Should it be
```python
if __debug__ and True:
```
?

Reply:
Oh yeah, I just enforces true to track down a bug